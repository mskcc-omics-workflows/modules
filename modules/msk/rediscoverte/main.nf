process REDISCOVERTE {
    tag "Sample(s) - ${[quant].flatten().size()}"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/rediscoverte:1.0.0':
        'docker.io/mskcc/rediscoverte:1.0.0' }"

    input:

    path quant
    path rmsk_annotation,                 stageAs: 'rollup_annotation/rmsk_annotation.RDS'
    path repName_repFamily_repClass_map,  stageAs: 'rollup_annotation/repName_repFamily_repClass_map.tsv'
    path genecode_annotation,             stageAs: 'rollup_annotation/GENCODE.V26.Basic_Gene_Annotation_md5.RDS'

    output:
    path "REdiscoverTE_rollup"            , emit: rollup
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def threads = task.cpus * 2

    """
    echo -e "sample\tquant_sf_path" > REdiscoverTE.tsv
    ls -1 *.quant.sf | awk -F. '{print \$1\"\\t\"\$0}' >> REdiscoverTE.tsv

    rollup.R \\
        --metadata=REdiscoverTE.tsv \\
        --datadir=rollup_annotation \\
        --nozero --threads=${threads} \\
        ${args} \\
        --outdir=REdiscoverTE_rollup

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rediscoverte: 1.0.0
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''

    """
    mkdir REdiscoverTE_rollup
    touch REdiscoverTE_rollup/abc.RDS

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rediscoverte: 1.0.0
    END_VERSIONS
    """
}
