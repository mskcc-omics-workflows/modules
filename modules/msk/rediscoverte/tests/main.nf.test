nextflow_process {

    name "Test Process REDISCOVERTE"
    script "../main.nf"
    process "REDISCOVERTE"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "rediscoverte"

    test("rna - fastq") {

        when {
            process {
                """
                input = [
                    file(params.test_data_mskcc['rediscoverte']['quant_sf'], checkIfExists: true),
                    file(params.test_data_mskcc['rediscoverte']['rmsk_annotation_subsampled_RDS'], checkIfExists: true),
                    file(params.test_data_mskcc['rediscoverte']['repName_repFamily_repClass_map_tsv'], checkIfExists: true),
                    file(params.test_data_mskcc['rediscoverte']['GENCODE_V26_Basic_Gene_Annotation_md5_RDS'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("rna - fastq - stub") {

        options "-stub"

        when {
            process {
                """

                input = [
                    file('quant_sf'),
                    file('rmsk_annotation_subsampled_RDS'),
                    file('repName_repFamily_repClass_map_tsv'),
                    file('GENCODE_V26_Basic_Gene_Annotation_md5_RDS')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
