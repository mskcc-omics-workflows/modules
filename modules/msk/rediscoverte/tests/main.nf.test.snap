{
    "rna - fastq - stub": {
        "content": [
            {
                "0": [
                    [
                        "abc.RDS:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2ca34b922d21d432437a6ca0e5546234"
                ],
                "rollup": [
                    [
                        "abc.RDS:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2ca34b922d21d432437a6ca0e5546234"
                ]
            }
        ],
        "timestamp": "2024-07-01T13:16:16.083084"
    },
    "rna - fastq": {
        "content": [
            {
                "0": [
                    [
                        "GENE_1_raw_counts.RDS:md5,f5ac77f8dcf054aec3836246fe2f5659",
                        "GENE_2_counts_normalized.RDS:md5,e7a7d004b033a0ddca28d24c4470ab60",
                        "GENE_3_TPM.RDS:md5,f5ac77f8dcf054aec3836246fe2f5659",
                        "RE_all_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_all_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_all_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_exon_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_exon_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intergenic_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intergenic_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intron_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intron_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2ca34b922d21d432437a6ca0e5546234"
                ],
                "rollup": [
                    [
                        "GENE_1_raw_counts.RDS:md5,f5ac77f8dcf054aec3836246fe2f5659",
                        "GENE_2_counts_normalized.RDS:md5,e7a7d004b033a0ddca28d24c4470ab60",
                        "GENE_3_TPM.RDS:md5,f5ac77f8dcf054aec3836246fe2f5659",
                        "RE_all_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_all_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_all_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_all_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_exon_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_exon_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_exon_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intergenic_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intergenic_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intergenic_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intron_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_repFamily_1_raw_counts.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8",
                        "RE_intron_repFamily_2_counts_normalized.RDS:md5,48d176fefee059f711fb56c3999118ca",
                        "RE_intron_repFamily_3_TPM.RDS:md5,01c79e6246f1e0f4ad927f0cc1f69ad8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2ca34b922d21d432437a6ca0e5546234"
                ]
            }
        ],
        "timestamp": "2024-07-01T13:16:09.400759"
    }
}